--- a/src/argtable3.c	2021-03-13 21:38:37.364733982 +0100
+++ b/src/argtable3.c	2021-03-13 21:37:09.740285831 +0100
@@ -40,6 +40,9 @@
 #include "argtable3.h"
 
 #define ARG_AMALGAMATION
+#ifdef _AIX
+#define ARG_REPLACE_GETOPT 1
+#endif
 
 /*******************************************************************************
  * argtable3_private: Declares private types, constants, and interfaces
@@ -1437,7 +1440,7 @@
 static const char illoptstring[] = "unknown option -- %s";
 #    endif
 
-#    ifdef _WIN32
+#    if defined(_WIN32) || defined(_AIX)
 
 /*
  * Windows needs warnx().  We change the definition though:
@@ -1468,6 +1471,8 @@
 #        if (defined(__STDC_LIB_EXT1__) && defined(__STDC_WANT_LIB_EXT1__)) \
             || (defined(__STDC_SECURE_LIB__) && defined(__STDC_WANT_SECURE_LIB__))
         _vsnprintf_s(opterrmsg, sizeof(opterrmsg), sizeof(opterrmsg) - 1, fmt, ap);
+#        elif defined(_AIX)
+        vsnprintf(opterrmsg, sizeof(opterrmsg) - 1, fmt, ap);
 #        else
         _vsnprintf(opterrmsg, sizeof(opterrmsg) - 1, fmt, ap);
 #        endif
